### DISCUSSION TITLE
Correctable design bug of the Sonoff NSPannel leads to TFT upload problems.

### DISCUSSION THREAD

**Discussion by gsieben** [Blackymas/NSPanel_HA_Blueprint#827] [General]
Hi guys, 

I have tried today for what feels like 1000 times to load the nspanel_eu.tft on the NSPanel. I had no success. The small nspanel_blank.tft was no problem. But the 6MB nspannel_eu.tft was not loadable. Sometimes the upload did not start at all. Then it ran fast for a short time and then became super slow. 

I noticed in the logs that there are always connection problems between the EPS and the Nexion display. Already at startup the first error messages of the serial connection appeared. 

Now I had the suspicion that maybe the power supply disturbs the serial connection between the ESP and the display. I could not establish the 9600 baud connection at all. 

To prevent this, I simply put an aluminum foil between the display and the power supply. And what should I say. The upload worked within less than 10 minutes. I have no error messages on the serial bus at all. 

If you have upload/serial error problems, then try this. It would be great if you give feedback. If this is generally the case, Blackymas could add it to the manual. Please make sure that the foil does not get between the connectors!

Cheers
Gabriel

![NSPanel with foil](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/2237858/0ccc95e7-cd36-465f-acab-79b6be7ee4ce)

**Comment by edwardtfn** [comment]
Thanks for sharing your journey.

> I could not establish the 9600 baud connection at all

In very rare occasions, when no valid TFT file is loaded, the speed will fall to 9600bps, but both the Sonoff settings and also nspanel_blank.tft, nspanel_eu.tft or nspanel_us.tft will all splicitly set the baud rate to 115200, so I will never suggest 9600bps. 😉

Are you using a local Web server?

**Reply by gsieben** [reply]
I just mentioned that about the 9600. The scientist got carried away with me. I suspect that the frequency bands overlap extremely here. But I have not tested it, because it is not relevant. The Nextion of course also works with 9600 ... See: https://esphome.io/components/display/nextion.html

I have used local and remote webserver. But this is also not relevant. As I said. The important thing is, that the errors on the serial bus are gone. And so the update was possible with full 115200 bps. At least on my device. But I only have one and can't say if it always works. But I guess yes.

**Reply by edwardtfn** [reply]
> The Nextion of course also works with 9600 ... See: https://esphome.io/components/display/nextion.html

Yes, Nextion supports a bunch of 'baud_rate' options, including 9600bps, which is a special one as it is a "fallback" rate, so if the display cannot stablish the communication in a higher frequency, it will try at 9600bps.
However, as I mentioned before, both Sonoff original firmware, but also ALL the TFT files from this project will explicitly define the baud rate to 115200bps. So, if everything fails, I would try reducing to 9600bps, but wift really low hope. Everything should always work at 115200bps.

> I have used local and remote webserver. But this is also not relevant. As I said. The important thing is, that the errors on the serial bus are gone. And so the update was possible with full 115200 bps. At least on my device. But I only have one and can't say if it always works. But I guess yes.

This is relevant. In the absolute maiority of the cases the issue is not related to the baud rate, but to issues related to the panel trying to get (download) the new TFT file from the server. Local server are way more reliable for this, however it requires more steps, which means more chances for a file transfer issue.

**Reply by gsieben** [reply]
I don't understand this people. 

This strange discussion misses the content of my message. Just as a side note. You are discussing with a physicist. 

According to my research, it is clearly due to the serial errors. It is to be read again and again that the uploading of the TFT took forever. And that is certainly not due to the Internet connection of a modern Internet connection. No matter how bad it is. It easily manages 115200bps. And - everyone can also run it on the local server! The people who use the NSPannel are certainly tecnisch skillful. 

So could you just leave it that those who have eternal problems with the upload of the tft can try it with the aluminum foil. And don't spam this post with unimportant infromation/discussion. I will say no more about it.

**Reply by edwardtfn** [reply]
Sorry for messing up your physicist-level post. 😜

**Comment by Blackymas** [comment]
thanks for sharing <3
